src/ser.rs

```diff
@@ -492,7 +504,21 @@ where
     where
         T: ?Sized + ser::Serialize,
     {
-        value.serialize(&mut **self)
+        use tagged::CborTag;
+        let t_fn: fn(&T, Self) -> Result<()> = T::serialize;
```

Comment:
This seems like an unreliable way of doing this detection. You could e.g. potentially end up with a situation where there are two copies of `CborTag::serialized` that ended up in different compilation units.